### PR TITLE
docs: new SHA-256 doc

### PR DESCRIPTION
**Pull request by Trivo25**
The SHA-2 hash function requires binary arithmetic and operates over binary data. SHA-2 is not native to most zero knowledge proofs.

Because SHA-256 operates over binary data instead of native Field elements like Poseidon, o1js uses the `Bytes` type. `Bytes` is a fixed-length array of bytes that can be used to represent binary data and is represented as an array of `UInt8` elements.

This doc defines how to use `Bytes` by extending the `Bytes` class and specifying the length of bytes.

### CODE REVIEW

**Comment by barriebyron** [comment]
hi @Trivo25 is this PR ready for review? (or did you want to convert this PR to a draft PR?)

Let me know and I will jump in when you're ready. Thank you!

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:2]:
```suggestion
title: SHA-256
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:4]:
```suggestion
description: A comprehensive guide on how to use SHA-2 hashes in o1js.
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:22]:
```suggestion
# SHA-256
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:25]:
```suggestion
It is also the predecessor of [Keccak (SHA-3)](/zkapps/o1js/keccak). The SHA-2 family is available with different output lengths, including 256-bit (SHA-256) or 512-bit (SHA-512). o1js supports only SHA-2 with an output length of 256 bits.
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:27]:
```suggestion
## What is SHA-2 and SHA-256?
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:30]:
```suggestion
widely used for traditional Web2 applications and protocols and blockchain technology. For example, Bitcoin's block headers are hashed twice using SHA256. 
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:33]:
```suggestion
## SHA-256 and Poseidon
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:38]:
```suggestion
In contrast, SHA-2 is a hash function that requires binary arithmetic. It operates over binary data and is not native to most zero knowledge proofs. For this reason, SHA-256 is not as efficient as Poseidon. However, it is still very useful for verifying Ethereum transactions and blocks.
So, when you choose what hash function to use, an important consideration is the use case and the data that needs to be hashed.
```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:48]:
```suggestion
SHA-256 is available in the following configuration under the `Hash` namespace in o1js:

- `Hash.SHA2_256`: SHA2-256 hash function with output size of 256 bits.

Because SHA-256 operates over binary data instead of native Field elements like Poseidon, o1js uses the `Bytes` type. `Bytes` is a fixed-length array of bytes that can be used to represent binary data.
Under the hood, `Bytes` is represented as an array of `UInt8` elements.

```

File: docs/zkapps/o1js/sha256.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/sha256.mdx:71]:
```suggestion
### SHA-256 - API reference
```

**Comment by Trivo25** [comment]
@barriebyron all comments have been addressed - the PR is ready for review and can be merged once approved and the o1js release went out

**Review by barriebyron — APPROVED** [review]
fabulous!

File: docs/zkapps/o1js/keccak.mdx
Comment (barriebyron) [inline docs/zkapps/o1js/keccak.mdx:111]: @Trivo25  is calling out SHA-256 and Keccak too detailed?

Comment (Trivo25) [reply docs/zkapps/o1js/keccak.mdx:111]: Nope, I think it reads well and makes sense!

**Pull request merged**
